src/include/catalog/postgres/pg_proc_impl.h

```diff
@@ -99,72 +107,83 @@ class PgProcImpl {
+   * @return                True if the procedure was successfully found and dropped. False otherwise.
    */
   bool DropProcedure(common::ManagedPointer<transaction::TransactionContext> txn, proc_oid_t proc);
 
   /**
-   * Set the procedure context pointer column of the specified procedure.
+   * @brief Set the procedure context pointer column of the specified procedure.
    *
-   * @return False if the proc_oid does not correspond to a valid procedure. True if the set was successful.
+   * @param txn             The transaction to use.
+   * @param proc            The OID of the procedure to set the context for.
+   * @param func_context    The procedure context to set to.
+   * @return                False if the proc_oid does not map to a valid procedure. True if the set was successful.
    *
    * TODO(WAN): See #1356 for a discussion on whether this should be exposed separately.
    */
   bool SetProcCtxPtr(common::ManagedPointer<transaction::TransactionContext> txn, proc_oid_t proc_oid,
                      const execution::functions::FunctionContext *func_context);
 
   /**
-   * Get the procedure context pointer column of the specified procedure.
+   * @brief Get the procedure context pointer column of the specified procedure.
    *
-   * @return The procedure context pointer of the specified procedure, guaranteed to not be nullptr.
+   * @param txn             The transaction to use.
+   * @param proc_oid        The OID of the procedure whose procedure context is to be obtained.
+   * @return                The procedure context pointer of the specified procedure, guaranteed to not be nullptr.
    */
   common::ManagedPointer<execution::functions::FunctionContext> GetProcCtxPtr(
       common::ManagedPointer<transaction::TransactionContext> txn, proc_oid_t proc_oid);
 
   /**
-   * Get the OID of a procedure from pg_proc.
+   * @brief Get the OID of a procedure from pg_proc.
    *
    * @param txn             The transaction to use.
    * @param dbc             The catalog that pg_proc is in.
    * @param procns          The namespace of the procedure to look in.
    * @param procname        The name of the procedure to look for.
    * @param arg_types       The types of all arguments in this function.
-   * @return The OID of the procedure if found. Else INVALID_PROC_OID.
+   * @return                The OID of the procedure if found. Else INVALID_PROC_OID.
    */
   proc_oid_t GetProcOid(common::ManagedPointer<transaction::TransactionContext> txn,
                         common::ManagedPointer<DatabaseCatalog> dbc, namespace_oid_t procns,
                         const std::string &procname, const std::vector<type_oid_t> &arg_types);
 
-  /** Bootstrap all the builtin procedures in pg_proc. */
+  /** @return True if the provided @p oid is within the builtin proc range. */
+  bool IsBuiltinProc(const proc_oid_t oid) const {
+    return lowest_builtin_proc_oid_ <= oid && oid < highest_builtin_proc_oid_;
+  }
```

Comment:
I don't think we need this.  As far as I know, the only legitimate reason to distinguish between builtins and UDFs is to hand-off to special-cased logic during recovery.

If we are in fact storing everything we need already in the catalog tables to handle all of our builtins without special logic during recovery, then that's great and we don't need this function.

If not, then we need the function but this implementation won't work because it is only accurate *after* calling bootstrap (which will never be called on a recovered database).

Reply:
Sounds legit.